src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/CommentEnteredSubscription.tsx

```diff
@@ -0,0 +1,266 @@
+  storyConnectionKey: string,
+  tag?: string
+) {
+  const rootField = store.getRootField("commentEntered");
+  if (!rootField) {
+    return;
+  }
+  const comment = rootField.getLinkedRecord("comment")!;
+  const commentsEdge = store.create(
+    `edge-${comment.getValue("id")!}`,
+    "CommentsEdge"
+  );
+  commentsEdge.setValue(comment.getValue("createdAt"), "cursor");
+  commentsEdge.setLinkedRecord(comment, "node");
+  const story = store.get(storyID)!;
+  const connection = ConnectionHandler.getConnection(
+    story,
+    storyConnectionKey,
+    {
+      orderBy: GQLCOMMENT_SORT.CREATED_AT_DESC,
+      tag,
+    }
+  )!;
+  const linked = connection.getLinkedRecords("viewNewEdges") || [];
+  connection.setLinkedRecords(linked.concat(commentsEdge), "viewNewEdges");
+}
+
+function updateForOldestFirst(
+  store: RecordSourceSelectorProxy<unknown>,
+  storyID: string,
+  storyConnectionKey: string,
+  tag?: string
+) {
+  const story = store.get(storyID)!;
+  const connection = ConnectionHandler.getConnection(
+    story,
+    storyConnectionKey,
+    {
+      orderBy: GQLCOMMENT_SORT.CREATED_AT_ASC,
+      tag,
+    }
+  )!;
+  const pageInfo = connection.getLinkedRecord("pageInfo")!;
+  // Should not be falsy because Relay uses this information to determine
+  // whether or not new data is available to load.
+  if (!pageInfo.getValue("endCursor")) {
+    // Set cursor to oldest date, to load from the beginning.
+    pageInfo.setValue(new Date(0).toISOString(), "endCursor");
+  }
+  pageInfo.setValue(true, "hasNextPage");
+}
+
+/**
+ * Returns depth until ancestor.
+ */
+function determineDepthTillAncestor(
+  comment: RecordProxy,
+  ancestorID?: string | null
+) {
+  let depth = 0;
+  let cur: RecordProxy | null | undefined = comment;
+  while (cur) {
+    cur = cur.getLinkedRecord("parent");
+    if (cur && cur.getValue("createdAt")) {
+      depth++;
+      // Stop when reaching base ancestor.
+      if (cur.getValue("id") === ancestorID) {
+        return depth;
+      }
+    } else if (depth === 0) {
+      return null;
+    }
+  }
+  return depth;
+}
+
+function insertReply(
+  store: RecordSourceSelectorProxy<unknown>,
+  liveDirectRepliesInsertion: boolean,
+  ancestorID?: string | null
+) {
+  const comment = store
+    .getRootField("commentEntered")!
+    .getLinkedRecord("comment")!;
+  const parentID = comment.getLinkedRecord("parent")!.getValue("id")! as string;
+  const parentProxy = store.get(parentID)!;
+
+  const depth = determineDepthTillAncestor(comment, ancestorID);
+  if (depth === null) {
+    // could not trace back to ancestor, discard.
+    return;
+  }
+
+  // Comment is just outside our visible depth.
+  if (depth >= 4) {
+    // Inform last comment in visible tree about the available replies.
+    // This will trigger to show the `Read More of this Conversation` link.
+    const replyCount = parentProxy.getValue("replyCount") || 0;
+    parentProxy.setValue((replyCount as number) + 1, "replyCount");
```

Comment:
Might be better to set the result of `.getValue` to a number instead?

```suggestion
    const replyCount = (parentProxy.getValue("replyCount") as number) || 0;
    parentProxy.setValue(replyCount + 1, "replyCount");
```